libraries/chain/asset_evaluator.cpp

```diff
@@ -391,27 +391,61 @@ void check_children_of_bitasset(database& d, const asset_update_bitasset_operati
 
             FC_ASSERT( !new_backing_asset.is_market_issued(),
                   "A non-blockchain controlled BitAsset would be invalidated by changing this backing asset.");
+
+            // Check if the new backing asset is itself backed by something. It must be CORE or a UIA
+            if ( new_backing_asset.is_market_issued() )
```

Comment:
* 4 lines above we've made sure that `new_backing_asset.is_market_issued()` is `false`.
* Actually the check for A-B-C-X (when changing A to be backed by B) should NOT be put in `check_children_of_bitasset`, since
  * when A has a child, actually we need to prevent CHILD-A-B-X (aka B can't be an MPA), so
  * it only makes sense to prevent A-B-C-X when A doesn't have a child;

Reply:
Moved the check outside of check_children_of_bitasset